### ISSUE TITLE
ames: Reusing port can cause forwarding loop

### ISSUE THREAD

**Issue by philipcmonk** [urbit/urbit#2690]
If you have a planet on port X and then you move it somewhere else and put its star on that same port, when you turn it on it'll flood itself and its galaxy with ames message to forward to the planet.

Not sure how best to mitigate this.  Ideally you want to say "don't send to myself", but we don't actually know our own lane.

**Comment by belisarius222** [comment]
We move star into planet's old lane and restart star. Star tries to talk to planet, so it sends packet to planet's old lane, which is now star's own lane. Packet comes back to star, which thinks it's supposed to forward it because packet says its receiver is planet, not star itself.

I think we can fix that easily. If we receive a packet where we were the sender, drop the packet.

But it seems star also sends packets to galaxy. I'm not sure what's going on there.

—
~rovnys-ricfer
https://urbit.org

**Comment by belisarius222** [comment]
To clarify, the fix I described would happen in the Ames vane, where we would check the @p of the sender, irrespective of lane.

—
~rovnys-ricfer
https://urbit.org

**Comment by philipcmonk** [comment]
We're not the sender, though. The sender is some unrelated ship.

**Comment by belisarius222** [comment]
Oh...

—
~rovnys-ricfer
https://urbit.org

**Comment by tomholford** [comment]
>If you have a planet on port X and then you move it somewhere else and put its star on that same port

Is it possible that a planet --> moon pairing could be affected by the same issue? I have several moons spawned from my planet. In the dojo for my planet, I see this periodically:
```
ames: forwarded 3435000000 total
```

And in the moon's dojo:
```
ames: packet dropped (3361311000 total)
```

**Comment by philipcmonk** [comment]
Yes, under some circumstances.  For now, I recommend not using the same port for different ships.

**Comment by philipcmonk** [comment]
You should never see those printfs from anything other than a galaxy

**Comment by Fang-** [comment]
Noting for the record that this may be mitigated by #5815.
